### PR TITLE
[8.x] Add extend option to Atomic Locks 

### PR DESCRIPTION
**Pull request by georgeboot**
This PR adds support for extending existing atomic locks.

Adapters to implement:

- [x] Redis
- [x] Memcached
- [ ] DynamoDB

## Use case

Imagine a support system, where only one agent at a time, can work on a case.

When implementing this with atomic locks, you would want to lock a case for 30 seconds, and have the page poll every 15 seconds. That poll request should be able to extend the existing lock. 

When the agent leaves the page, the case will automatically be unlocked as soon as the lock expires.

## Help wanted
I'm not experiences with DynamoDB, so I can't figure out how to easily extend the TTL of a record. Perhaps someone else wishes to share their wisdom?

### CODE REVIEW

File: src/Illuminate/Cache/Lock.php [hunk 1]
```diff
@@ -58,6 +58,14 @@ public function __construct($name, $seconds, $owner = null)
      */
     abstract public function acquire();
 
+    /**
+     * Extend the lock.
+     *
+     * @param int $seconds
+     * @return bool
+     */
+    abstract public function extend($seconds);
```
Comment (GrahamCampbell) [inline src/Illuminate/Cache/Lock.php:67]: This is a major breaking change, so this PR will have to target the master branch.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Cache/Lock.php (hunk 1 above)
Comment (georgeboot) [reply src/Illuminate/Cache/Lock.php:67]: Sorry, missed that. Updated to master.

File: src/Illuminate/Cache/RedisLock.php [hunk 2]
```diff
@@ -41,14 +41,25 @@ public function acquire()
...
 
+    /**
+     * Extend the lock.
+     *
+     * @param int $seconds
+     * @return bool
+     */
+    public function extend($seconds)
+    {
+        return (bool)$this->redis->expire($this->name);
```
Comment (taylorotwell) [inline src/Illuminate/Cache/RedisLock.php:52]: How does this extend anything? The seconds are literally never used.

File: src/Illuminate/Cache/RedisLock.php [hunk 3]
```diff
@@ -41,14 +41,25 @@ public function acquire()
...
+
     /**
      * Release the lock.
      *
      * @return bool
      */
     public function release()
     {
-        return (bool) $this->redis->eval(LuaScripts::releaseLock(), 1, $this->name, $this->owner);
+        return (bool)$this->redis->eval(LuaScripts::releaseLock(), 1, $this->name, $this->owner);
```
Comment (taylorotwell) [inline src/Illuminate/Cache/RedisLock.php:62]: Why?

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, **tests**, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Comment by georgeboot** [comment]
Thanks for the feedback. Will re-submit when it’s more complete.

@taylorotwell What is the best way for me to get someone collaborating on the DynamoDB part?

**Pull request closed without merging**